tests/README.md

```diff
@@ -0,0 +1,202 @@
+  `test_name`.
+- Various `zassert_*` [assertions] can then be used to check the equality, if
+  condition is true, false, etc. Common theme is that they all can take a
+  message with arguments in the end (just like `printf`), which is printed if
+  assert fails.
+- A suit of tests can have common setup, teardown, etc. functions to decrease
+  boilerplate code. See [Creating a test suite] section for more info.
+
+See
+[zephyr/tests/ztest/base](https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/ztest/base)
+for some basic usecases of Ztest framework.
+
+[assertions]:
+  https://docs.zephyrproject.org/latest/develop/test/ztest.html#assertions
+[creating a test suite]:
+  https://docs.zephyrproject.org/latest/develop/test/ztest.html#creating-a-test-suite
+
+## Twister
+
+Twister is a script that scans for the set of unit test applications in the git
+repository and attempts to execute them. For listed boards it can build the
+applications, run them, check if they every successful and generate a report.
+
+Due to the number of features that it has it can take sometime to figure out how
+to do something that you want, see east twister `--extra-help` for the supported
+options.
+
+Also see
+[Zephyr's documentation](https://docs.zephyrproject.org/latest/develop/test/twister.html)
+for more info.
+
+### Simple example
+
+From project's root directory run:
+
+```shell
+east twister -b native_posix -T tests
+```
+
+Twister will detect all projects with `testcase.yaml` in their project root
+under `tests` directory and execute them.
+
+To run the same thing on the target you need to provide more info to the
+Twister:
+
+```shell
+east twister -T tests -p nrf52dk_nrf52832 --device-testing --device-serial /dev/ttyACM0
+```
+
+Check the output for the results.
+
+### `twister-out` folder
+
+With every invocation of the `east twister` a `twister-out` folder is created.
+It contains all build folders of the created projects, as well as the report
+files.
+
+Keep in mind, if you keep running `east twister` command and not deleting the
+`twister-out` folder, then command will rename it to `twister-out.X` (Where `X`
+is a unique ascending number, starting with 1) before running the full build.
+
+### `testcase.yaml` file
+
+`testcase.yaml` file marks that a Twister should pick up the test folder and
+build it. Its content defines for which platforms it should be build and tested.
+
+See [Test Cases] section for more info.
+
+[test cases]:
+  https://docs.zephyrproject.org/latest/develop/test/twister.html#test-cases
+
+### Coverage report
+
+Twister supports creating coverage reports from test runs with `gcovr` tool.
+Adding extra flags to the command invocation creates a html file (among others)
+which shows code coverage for each file.
+
+```shell
+east twister -T tests -p native_posix --coverage  --coverage-tool gcovr --coverage-basedir lib
+```
+
+Note that you might need to change `--coverage-basedir` or add new ones for your
+project.
```

Comment:
Is there a way to run code coverage on `zephyr.elf `directly?
You have _run directly_ and _debug_ instructions at line 55. Can you add instructions how to run code coverage directly there as well?

Reply:
There is a way: https://docs.zephyrproject.org/latest/develop/test/coverage.html#coverage-reports-using-the-posix-architecture

But I guess I could just link this? Or just leave it be?